### ISSUE TITLE
scipy fails to build on Travis

### ISSUE THREAD

**Issue by certik** [sympy/sympy#9511]
Many times scipy fails to build on Travis, e.g.:
https://travis-ci.org/sympy/sympy/jobs/65920076

```
    Post-processing (stage 2)...

    Building modules...

        Building module "mvn"...

            Constructing wrapper function "mvnun"...

              value,inform = mvnun(lower,upper,means,covar,[maxpts,abseps,releps])

            Constructing wrapper function "mvndst"...

              error,value,inform = mvndst(lower,upper,infin,correl,[maxpts,abseps,releps])

            Constructing COMMON block support for "dkblck"...

              ivls

        Wrote C/API module "mvn" to file "build/src.linux-x86_64-3.4/scipy/stats/mvnmodule.c"

        Fortran 77 wrappers are saved to "build/src.linux-x86_64-3.4/scipy/stats/mvn-f2pywrappers.f"

      adding 'build/src.linux-x86_64-3.4/fortranobject.c' to sources.

      adding 'build/src.linux-x86_64-3.4' to include_dirs.

      adding 'build/src.linux-x86_64-3.4/scipy/stats/mvn-f2pywrappers.f' to sources.

    building extension "scipy.ndimage._nd_image" sources

    building data_files sources

    build_src: building npy-pkg config files

    Running from scipy source directory.

    /home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/numpy/distutils/system_info.py:495: UserWarning:

        UMFPACK sparse solver (http://www.cise.ufl.edu/research/sparse/umfpack/)

        not found. Directories to search for the libraries can be specified in the

        numpy/distutils/site.cfg file (section [umfpack]) or by setting

        the UMFPACK environment variable.

      warnings.warn(self.notfounderror.__doc__)

Installing collected packages: scipy

  Running setup.py install for scipy

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```

This needs to be fixed. We should install a binary package using Anaconda. This is critical, because it breaks many of our test runs. It only happens sometimes.

**Comment by certik** [comment]
https://travis-ci.org/sympy/sympy/jobs/65926290

**Comment by certik** [comment]
https://travis-ci.org/sympy/sympy/jobs/66260656

**Comment by moorepants** [comment]
If you use the `system_site_packages` flag and install SciPy from apt may be the simplest solution. Getting our whole testing infrastructure working with conda will take a bit more work. This could be a nice SciPy sprint idea. We tried it last year:

#7404

**Comment by moorepants** [comment]
People tried to add SciPy on travis in these too: #2883 and #2696.

**Comment by moorepants** [comment]
How doe it break lots of test runs if it is only installed in the Theano test?

**Comment by toolforger** [comment]
> This needs to be fixed.

I agree this should be fixed.

 > We should install a binary package using Anaconda.

What's the advantage over using Travis' directory persistency?

**Comment by toolforger** [comment]
Okay, I did a bit of reading on Travis' persistency mechanisms.
TL;DR: It most likely won't work for us.

Issue is that caching is available only for "new infrastructure" (which is also described as "faster").
Which does not support "sudo", which we need to install some dependencies.
Now they did offer an alternate, sudo-less way to install packages (specify a list), *but* the PPAs and packages need to be whitelisted, and we've got a couple of them which aren't on the whitelist (mostly TeX stuff and the sagemath PPA). And they are asking people to be patient if they want more stuff whitelisted... so Travis' caching is not a short-term solution.

**Comment by toolforger** [comment]
@certik As far as I could compare your example above to logs of successful builds, it seems that everything up to and including the UMFPACk warning is normal.
Seems like scipy's setup.py simply hangs.

I dimly recall this having been observed earlier, with speculation that maybe it's a stalled download inside of setup.py, or something else that should time out and doesn't.

**Comment by certik** [comment]
It breaks a single Theano test in all the test runs. Well, about 50% or so.

**Comment by davidlowryduda** [comment]
This doesn't seem to be a problem anymore, right? I think this issue can now be closed.

**Comment by oscarbenjamin** [comment]
This isn't an issue now. Closing
